updating pre-commit and adding pyproject.toml

GuillermoFidalgo opened the pull request.

GuillermoFidalgo committed:
add pyproject.toml and change pre-commit

GuillermoFidalgo committed:
delete file

GuillermoFidalgo committed:
add licensing text

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

GuillermoFidalgo committed:
Update pyproject.toml

mambelli commented on .pre-commit-config.yaml:
```diff
@@ -20,18 +20,8 @@ repos:
       - id: trailing-whitespace
         args:
           - "--markdown-linebreak-ext=md"
-  - repo: "https://github.com/pre-commit/pygrep-hooks"
-    rev: v1.9.0
-    hooks:
-      - id: python-check-blanket-noqa
-      - id: python-check-blanket-type-ignore
-      - id: python-use-type-annotations
-  - repo: "https://github.com/pycqa/isort"
-    rev: 5.10.1
-    hooks:
-      - id: isort
```
Are these checks already included in ruff?

mambelli commented on .pre-commit-config.yaml:
```diff
@@ -49,14 +39,8 @@ repos:
+    rev: "v0.4.5"
     hooks:
-      - id: pylint
-  - repo: "https://github.com/fsfe/reuse-tool"
-    rev: v0.14.0
-    hooks:
-      - id: reuse
-        additional_dependencies:
-          - python-debian==0.1.40
```
Why did you remove the license checking?

mambelli commented:
Approving to get the repo as you like for the class. I may have to check and revert some changes in the precommit later

GuillermoFidalgo commented:
Sorry I didn't have a chance to answer the message. Today as we expected was very hectic.

I simply deleted some pre-commits and checked what worked. I remember reading a problem in the logs with isort. So I removed that because it is in ruff. 
But I remember that the licence check see was still working. At least it required the license on the toml file I added so I don't know what it does.


It could just as well be reimplemented.
